### DISCUSSION TITLE
Initial Modification of the Solution Matrix by Removal of BCs

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#27182] [Q&A Modules: General]
Given a typical 2nd order Poisson problem.
Au = f,

Where the discrete problem reads as,
Find u_h in V_h such that a_h(u_h,rho) = (f_h,rho) for all rho in V_h.

 A_hU_h = F_h 
Find u in V such that a(u,rho) = (f,rho) for all rho in V.

The continuous problem is defined with the boundary conditions, B(u) = g on the surface of the domain "cap_gamma". 

I would like to initialize Matrix A_h (and also U_h andF_h)  to not incorporated any BCs, corresponding to natural BCs.
S_h(cap_gamma). S_h(cap_gamma) denotes degrees of freedom of the vector U_h that are related to the boundary cap_gamma. Is it possible to modify the MOOSE framework code to accomplish this type of modification. If so what is a good starting point to perform the modification?

**Comment by GiudGiud** [comment]
Hello

You can turn on/off boundary conditions using the Controls system.
A common object to use for that purpose is this one:
https://mooseframework.inl.gov/source/controls/ConditionalFunctionEnableControl.html

Guillaume

**Reply by TLWise** [reply]
@GiudGiud Awesome! Thanks for pointing me in the right direction of the path of least resistance.
